File: fizzbuzz/fizzbuzz.go
/**
 * FizzBuzz
 *
 * Motivated by a Tom Scott's video on the subject.
 * https://youtu.be/QPZ0pIK_wsc
 *
 * Andre Zunino <neyzunino@gmail.com>
 * 15 August 2019
 */

package main

import (
    "fmt"
    "strings"
    "strconv"
)

func FizzBuzz(limit int) {
    var buffer strings.Builder
    buffer.Grow(8)
    for i := 1; i <= limit; i++ {
        if i % 3 == 0 {
            buffer.WriteString("Fizz")
        }
        if i % 5 == 0 {
            buffer.WriteString("Buzz")
        }
        if buffer.Len() == 0 {
            buffer.WriteString(strconv.Itoa(i))
        }
        fmt.Println(buffer.String())
        buffer.Reset()
    }
}

func main() {
    FizzBuzz(50)
}

File: fizzbuzz/fizzbuzz.rs
// FizzBuzz
//
// Andre Zunino <neyzunino@gmail.com>
// 16 April 2021

fn fizzbuzz(n: u32) {
    for i in 1..=n {
        let by_3 = i % 3 == 0;
        let by_5 = i % 5 == 0;
        if !by_3 && !by_5 {
            println!("{}", i);
            continue;
        }
        if by_3 {
            print!("Fizz");
        }
        if by_5 {
            print!("Buzz");
        }
        println!();
    }
}

fn main() {
    fizzbuzz(50);
}

